src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -67,11 +69,30 @@ public boolean doPrompts(Document document) {
 
         // Handle custom pre-rules:
         preRulesOK &= this.checkCheckStubTextOverflow((DisbursementVoucherDocument) document);
+        
+        setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(document);
 
         return preRulesOK;
     }
 
-    /**
+	private void setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(Document document) {
+		DisbursementVoucherDocument dvDoc = (DisbursementVoucherDocument) document;
+		DisbursementVoucherPayeeDetail dvPayeeDetail = dvDoc.getDvPayeeDetail();
+		
+		String payeeTypeCode = dvPayeeDetail.getDisbursementVoucherPayeeTypeCode();
+		String paymentReasonCode = dvPayeeDetail.getDisbVchrPaymentReasonCode();
+		Integer vendorHeaderId = dvPayeeDetail.getDisbVchrVendorHeaderIdNumberAsInteger();
+
+		if (getCuDisbursementVoucherTaxService().isForeignVendorAndTaxReviewNotRequired(payeeTypeCode,paymentReasonCode, vendorHeaderId)) {
+			dvDoc.getDvNonResidentAlienTax().setIncomeClassCode(DisbursementVoucherConstants.NRA_TAX_INCOME_CLASS_NON_REPORTABLE);
```

Comment:
I am ok with this code being in the PreRules. It does seem consistent with what pre-rules do in other cases such as DisbursementVoucherDocumentPreRules.checkSpecialHandlingIndicator().